### ISSUE TITLE
RELEASES: Version 2.10.1

### ISSUE THREAD

**Issue by glhays** [hassanhabib/The-Standard#272]
REQUIRES: #271

**Comment by glhays** [comment]
With the new version 2.10.1, we adhere to a full anemic exception model and move the string messaging to the Services layer as will be shown later. This new upgrade to the Standard will still enforce a strong binding to the exception messages and the testing messages. This will also put all string messages in the appropriate service layer logic.

### V2.10.0 Exception Model Sample:
```
public class FailedFileServiceException : Xeption
    {
        public FailedFileServiceException(Exception innerException)
            : base(message: "Failed file service error occurred, contact support.",
                innerException: innerException)
        { }

        public FailedFileServiceException(string message, Exception innerException)
            : base(message, innerException)
        { }
}
```
### V2.10.1 Exception (anemic) Model:
```
public class FailedFileServiceException : Xeption
    {
        public FailedFileServiceException(string message, Exception innerException)
            : base(message, innerException)
        { }
}
```
### V2.10.0...2.10.1 Test logic remains the same
```
[Fact]
        private void ShoudThrowServiceExceptionOnWriteIfServiceErrorOccurs()
        {
            // given
            string somePath = GetRandomString();
            string someContent = GetRandomString();
            var serviceException = new Exception();

            var failedFileServiceException =
                new FailedFileServiceException(
                    message: "Failed file service error occurred, contact support.",
                    innerException: serviceException);

            var fileServiceException =
                new FileServiceException(
                    message: "File service error occurred, contact support.",
                    innerException: failedFileServiceException);

            this.filesBrokerMock.Setup(broker =>
                broker.WriteToFile(It.IsAny<string>(), It.IsAny<string>()))
                    .Throws(serviceException);

            // when
            Action writeToFileAction = () =>
                this.fileService.WriteToFile(somePath, someContent);

            // then
            FileServiceException actualFileServiceException =
                Assert.Throws<FileServiceException>(writeToFileAction);

            actualFileServiceException.Should().BeEquivalentTo(
                fileServiceException);

            this.filesBrokerMock.Verify(broker =>
                broker.WriteToFile(It.IsAny<string>(), It.IsAny<string>()),
                    Times.Once);

            this.filesBrokerMock.VerifyNoOtherCalls();
        }
```
### V2.10.1 Services layer message handling examples:
#### Messages will now be within the service layer exception sub class like so.
```
Services/Foundations/Files/FileService.Exceptions.cs
```
```
 public partial class FileService
    {
        private delegate void ReturningNothingFunction();
        private static void TryCatch(ReturningNothingFunction returningNothingFunction)
        {
            try
            {
                returningNothingFunction();
            }
            catch (InvalidFilePathException invalidFilePathException)
            {
                throw CreateFileValidationException(invalidFilePathException);
            }
            catch (InvalidFileContentException invalidFileContentException)
            {
                throw CreateFileValidationException(invalidFileContentException);
            }
            catch (ArgumentNullException argumentNullException)
            {
                var invalidFileException =
                    new InvalidFileException(
                        message: "Invalid file error occurred.",
                        innerException: argumentNullException);

                throw CreateFileDependencyValidationException(invalidFileException);
            }
            catch (ArgumentException argumentException)
            {
                var invalidFileException =
                    new InvalidFileException(
                        message: "Invalid file error occurred.",
                        innerException: argumentException);

                throw CreateFileDependencyValidationException(invalidFileException);
            }
            catch (SerializationException serializationException)
            {
                var failedFileSerializationException =
                    new FailedFileSerializationException(
                        message: "Failed file serialization error occurred, contact support.",
                        innerException: serializationException);

                throw CreateFileDependencyException(failedFileSerializationException);
            }
            catch (Exception exception)
            {
                var failedFileServiceException =
                    new FailedFileServiceException(
                        message: "Failed file service error occurred, contact support.",
                        innerException: exception);

                throw CreateFileServiceException(failedFileServiceException);
            }
        }
        
         private static FileValidationException CreateFileValidationException(
            Xeption innerException)
        {
            return new FileValidationException(
                message: "File validation error occurred, fix the errors and try again.",
                innerException: innerException);
        }

        private static FileDependencyValidationException CreateFileDependencyValidationException(
            Exception innerException)
        {
            return new FileDependencyValidationException(
                message: "File dependency validation error occurred, fix the errors and try again.",
                innerException: innerException);
        }
C
        private static FileDependencyException CreateFileDependencyException(
            Xeption innerException)
        {
            return new FileDependencyException(
                message: "File dependency error occurred, contact support.",
                innerException);
        }

        private static FileServiceException CreateFileServiceException(
            Xeption innerException)
        {
            return new FileServiceException(
                message: "File service error occurred, contact support.",
                innerException: innerException);
        }
```

**Comment by glhays** [comment]
Closed via the release of `v2.10.1`
